### ISSUE TITLE
Hovering over a link loads CSS that affects the current page

### ISSUE THREAD

**Issue by BMorearty** [sveltejs/kit#8867]
### Describe the bug

Under some conditions, hovering over a link preloads CSS that modifies the appearance of the current page.

https://user-images.githubusercontent.com/16927/216515321-3fecd177-0e28-4f24-8618-a4a9ec8f4056.mov

## Preconditions

1. `data-sveltekit-preload-data="hover"` is set.
2. Either:
    1. CSS is loaded using an `import` statement in the `<script>` section.
    2. Or there are `:global` CSS tags in the `<style>` section.

I discovered this because I was using the `@html` tag to load Markdown-rendered HTML and I didn't want Svelte to remove the unused styles.

The only workaround I know of is for people not to use `data-sveltekit-preload-data="hover"`.

### Reproduction

1. clone this repo: https://github.com/BMorearty/sveltekit-css-load-bug
2. `pnpm install`
3. `pnpm dev`
4. Hit 'o' to open the browser
5. Hover over the link on the page and watch the h1 grow in size when the CSS for the other page is loaded.

### System Info

```Shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1
    Memory: 148.63 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.12.1/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 109.0
    Safari: 16.2
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.2
    @sveltejs/kit: ^1.0.0 => 1.3.10
    svelte: ^3.54.0 => 3.55.1
    vite: ^4.0.0 => 4.1.1
```

### Severity

serious, but I can work around it

**Comment by gtm-nayan** [comment]
See #5022 and #2783

Global styles, ie. ones marked with :global() or imported from a css file will stay on your page until your page is reloaded. Preloading another page's component preloads the css file as well making it affect your current page. 

But you'd have to defend against unwanted styles with either css modules or another scoping mechanism even if you weren't preloading for when your users navigate back from the page that applies those styles.

**Comment by enBonnet** [comment]
Do we have a section in the documentation to clarify this behavior?

**Comment by Wtower** [comment]
This also affects me. I understand from your comments why this is happening, but could you please elaborate on what scoping mechanisms are available to deal with this?

If I understand correctly, to use css modules I would have to rename bundled css files from other packages to `.module.css`.?

I suppose another way would be to import them with `?url` and append them to head `onMount` as with legacy JS, but this again feels dirty.

What would you recommend to do with bundled css files?

**Comment by chazzmoney** [comment]
I also experienced this problem. Some pages had a +layout.svelte file that loaded a css file.

My workaround had to include both suggestions from @BMorearty and @Wtower:
- remove data-sveltekit-preload-data="hover"
- add data-sveltekit-reload to the 'a' tags for pages where said css would conflict

I realize that the best solution is to migrate all css into the svelte mechanism, but it wasn't feasible at the time.

**Comment by ghostdevv** [comment]
For me setting `data-sveltekit-preload-data="off"` works

**Comment by hyunbinseo** [comment]
For imported CSS files, I am using this workaround:

```svelte
<!-- src/routes/(template)/+layout.svelte -->

<script lang="ts">
  import printFriendly from '/node_modules/print-friendly/dist/print-friendly.css?url';
</script>

<svelte:head>
  <link rel="stylesheet" href={printFriendly} />
</svelte:head>
```

Preload the CSS file in a appropriate place to avoid layout shift when navigating.

```svelte
<!-- src/routes/+layout.svelte -->

<script>
  import printFriendly from '/node_modules/print-friendly/dist/print-friendly.css?url';
</script>

<svelte:head>
  <link rel="preload" href={printFriendly} as="style" />
</svelte:head>
```

With this approach, anchor preload option does not have to be altered.

**Comment by steve-taylor** [comment]
TL;DR: Don't use Svelte's `<style>` block. It's catastrophically bad.

Off the back of this bug, I discovered that Svelte "by design" (i.e. too hard to do the right thing) never unloads CSS once it has been loaded. Combine this with selectors that use `:global` to select child DOM elements inside child components, and hover-to-preload, and CSS becomes an absolute nightmare to debug because it does all sorts of weird unexpected stuff based on the pages you had previously viewed and the order in which you viewed them. I was getting weird CSS layout issues and when I realised Svelte wasn't unloading CSS, I finally understood why I thought I was losing my marbles.

Faced with the possibility of using Tailwind in favour of Svelte's built-in CSS management, I elected to port a year's worth of SvelteKit code to Next.js with styled-components, which is considerate enough to tidy up after itself. As I convert these side-by-side, I'm noticing React is only slightly more verbose than Svelte for 95% of my components and that's on Svelte 4. When Svelte 5 lands, it'll flush the best parts of the Svelte DX down the toilet, such as `$:` and `<slot />`.

**Comment by steve-taylor** [comment]
I'm afraid to say that the only viable workaround for Svelte's poor CSS management is to use something like Tailwind, because it ties the lifetime of element styles to the elements themselves, whereas Svelte's CSS management never unloads CSS ~because it's too hard~ by design.

**Comment by hyunbinseo** [comment]
@Wtower this is a basic usage of CSS Modules in SvelteKit (Vite)

```css
/* src/routes/card.module.css */

article.card {
  height: 50px;
  width: 50px;
  background-color: yellow;
}

article.card::after {
  content: 'Hey!';
}
```

```svelte
<!-- src/routes/+page.svelte -->
<script>
  import cardStyles from './card.module.css';
</script>

<article class={cardStyles.card}></article>
<div class={cardStyles.card}></div>
```

The `article.card` style is only applied to the `<article>` element:

```html
<!-- Rendered HTML -->
<article class="_card_805h2_3"></article>
<div class="_card_805h2_3"></div>
```

```css
/* Generated CSS */
article._card_805h2_3 {
  height: 50px;
  width: 50px;
  background-color: yellow;
}
```

---

To apply styles to a Svelte component: (e.g. `+page.svelte`)

```css
/* src/routes/styles.module.css */
.root {
  > article {
    height: 50px;
    width: 50px;
    background-color: yellow;
  }
  > article::after {
    content: 'Hey!';
  }
}
```

```svelte
<!-- src/routes/+page.svelte -->
<script>
  import styles from './styles.module.css';
</script>

<!-- Limitation: wrapper element is needed. -->
<div class={styles.root}>
  <article></article> <!-- styled -->
</div>
```

This can be used instead of the direct CSS import. (e.g. `import './app.css`)

**Comment by eltigerchino** [comment]
Closing as duplicate of https://github.com/sveltejs/kit/issues/12933
